wodles/aws/subscribers/s3_log_handler.py

```diff
@@ -305,3 +305,101 @@ def process_file(self, message_body: dict) -> None:
         for event in events_in_file:
             self.send_msg(event, dump_json=False)
         aws_tools.debug(f'{len(events_in_file)} events sent to Analysisd', 2)
+
+
+class AWSSecurityHubSubscriberBucket(AWSSubscriberBucket):
+    """Class for processing events from AWS S3 buckets containing Security Hub related logs."""
+
+    @staticmethod
+    def _add_event_type_fields(details: dict, event: dict):
+        """
+        Add the corresponding fields into the event if the details contain them.
+
+        Parameters
+        ----------
+        details : dict
+            Source dictionary containing the events from the log file.
+        event : dict
+            Destination dictionary to be added to the event sent to Wazuh.
+        """
+        key_actions = {
+            'findings': lambda source, destination: destination.setdefault('finding', source['findings'][0]),
+            'actionName': lambda source, destination: destination.setdefault('actionName', source['actionName']),
+            'actionDescription': lambda source, destination: destination.setdefault('actionDescription',
+                                                                                    source['actionDescription']),
+            'insightName': lambda source, destination: destination.setdefault('insightName', source['insightName']),
+            'insightArn': lambda source, destination: destination.setdefault('insightArn', source['insightArn']),
+            'resultType': lambda source, destination: destination.setdefault('resultType', source['resultType']),
+            'insightResults': lambda source, destination: destination.setdefault('insightResults',
+                                                                                 source['insightResults'])
```

Comment:
I think this could be simplified by declaring a function like this

```py
def action(details: dict, event: dict, field_name: str) -> None:
    event.setdefault(field_name, details[field_name])
```

And then using it to populate the dictionary

```py
'actionName': action,
'actionDescription': action,
'insightName': action,
'insightArn': action,
'resultType': action,
'insightResults': action,
```

Or even populate it using an array of field names

```py
for key in fields.items():
    if key in details:
        action(details, event, key)
```

Reply:
After applying the suggested changes, the events remain being correctly processed:
<details>

![image](https://github.com/wazuh/wazuh/assets/26691145/9f7acd88-d730-44e1-bdb1-0f63f7dd6f59)


</details>